Utils/github_workflow_scripts/handle_external_pr.py

```diff
@@ -253,35 +255,124 @@ def is_requires_security_reviewer(pr_files: list[str]) -> bool:
+    """
+    Checks if tim reviewer needed, if the pack is new and not part of Master.
+    First the remote branch is going to be checked out and then verified for the data
+
+    Arguments:
+        - `pr_files`: ``List[str]``: The list of files changed in the Pull Request. Will be used to determine
+        whether a security engineer is required for the review.
+        - 'external_pr_branch': str : name of branch to checkout
+        - 'repo_name': str : name of repository
+
+    Returns: `bool` whether a security engineer should be assigned
+    """
+    try:
+        fork_owner = os.getenv('GITHUB_ACTOR')
+        with Checkout(
+            repo=Repo(Path().cwd(), search_parent_directories=True),
+            branch_to_checkout=external_pr_branch,
+            # in marketplace contributions the name of the owner should be xsoar-contrib
+            fork_owner=fork_owner if fork_owner != 'xsoar-bot' else 'xsoar-contrib',
+            repo_name=repo_name
+        ):
+            for file in pr_files:
+                if 'CONTRIBUTORS.json' in file or 'Author_image' in file or 'README.md' in file or ".pack-ignore" in file:
+                    continue
+                content_object = BaseContent.from_path(CONTENT_PATH / file)
+                is_tim_needed = check_if_pack_or_integration_is_feed(content_object)
+                if is_tim_needed:
+                    return True
+    except Exception as er:
+        print(f"couldn't checkout branch to get metadata, error is {er}")
+        # if the checkout didn't work for any reason, will try to go over files manually
+        return check_files_of_pr_manually(pr_files)
+    return False
+
+
+def is_tim_content(pr_files: list[str], external_pr_branch: str, repo_name: str) -> bool:
+    """
+    This is where the actual search for feed:True or relevant tags or categories are being searched
+    according to the login in is_tim_reviewer_needed.
+    It also taking into account if the pack exist in Master or it is new and need to checkout remote branch
+
+    Arguments:
+    - pr_files: List[str] The list of files changed in the Pull Request.
+    - 'external_pr_branch' (str) : name of the external branch to checkout
+    - 'repo_name' (str) : name of the external repository
+
+    Returns: returns True or False if tim reviewer needed
+    """
+    for file in pr_files:
+        if 'CONTRIBUTORS.json' in file or 'Author_image' in file or 'README.md' in file or ".pack-ignore" in file:
+            continue
+        content_object = BaseContent.from_path(CONTENT_PATH / file)
+        if not content_object:
+            # This means we were not able to find the file in content repo, and the contribution is new
+            print(f'for file {file}, the pack doesn\'t exist in Master and going to be searched in the remote branch')
+            return check_new_pack_metadata(pr_files, external_pr_branch, repo_name)
```

Comment:
Why aren't we always checking out the contributor repo? it is more accurate...
Also, you are calling check_new_pack_metadata for every file, this should not be the case it should only be called once...

Reply:
Basically the logic of the check_new_pack_metadata function should be the logic of the is_tim_content

Reply:
is_tim_content is only doing the:
BaseContent.from_path(CONTENT_PATH / file)
and based on the results it chooses whether to go to check_new_pack_metadata - which will checkout the remote branch if this is a new pack and not part of the master.

or if this is in the master and the from_path returned an item - I will go to check_if_item_is_tim

you're correct about running for each file, I've added break after.
about checking out every time - I think it saves time that if we don't need to checkout (if this is not a new pack - since most of the PR's are not new) then no need to go there.